### DISCUSSION TITLE
Will pywebview work on a Raspberry Pi?

### DISCUSSION THREAD

**Discussion by valnurat** [r0x0r/pywebview#1155] [Q&A]
Hi

Just a small question if pywebview will run on a Raspberry Pi, because I have tried to run it.

I installed the pywebview by doing this:
- pip install pywebview
- sudo apt install python3-gi python3-gi-cairo gir1.2-gtk-3.0 gir1.2-webkit2-4.0

But as you can see in the attached picture I don't see anything. Only a window but not the webpage.

![Skærmbillede 2023-06-14 222620](https://github.com/r0x0r/pywebview/assets/76236110/472c29cd-394f-43d3-bcc5-4a9d5dd73a21)

Do I need to do more or is pywebview not for raspberries?
Br
Valnurat

**Comment by r0x0r** [comment]
It should be possible, but with a limited success. See this old thread for possible problems https://github.com/r0x0r/pywebview/issues/523

I have not tried pywebview on RPI since then, so no idea what the current situation is.

**Reply by valnurat** [reply]
Ok. I looked at the link, but I it don't solve my issue.

**Reply by r0x0r** [reply]
Sorry I cannot help you. I don't have a spare RPI at the moment and it has been a while since I tested it on RPI myself. 
My opinion that it should work as long as you install correct dependencies. Have you tried QT?

**Reply by valnurat** [reply]
Could you update me how I should install the QT, please? And should I just install it even though I installed:

- sudo apt install python3-gi python3-gi-cairo gir1.2-gtk-3.0 gir1.2-webkit2-4.0

Btw, was is QT?

**Reply by valnurat** [reply]
I solved the issue.
There's something wrong when I VNC to the raspberry as headless.
If I connect a monitor it shows the standard example, but unfortunately pywebview can't view www.sas.dk properly. Some infomation is missing vs if you use any other browser.

**Reply by r0x0r** [reply]
See https://pywebview.flowrl.com/guide/installation.html#dependencies for QT installation

Poor rendering is probably because of the outdated WebKit renderer shipped with GTK. You have to figure out how to update it to a more recent version.

**Comment by paulocoutinhox** [comment]
It works, you can get all instructions and sample here:
https://github.com/paulocoutinhox/pyinstaller-sample
